packages/aio-commerce-lib-core/source/headers/auth.ts

```diff
@@ -0,0 +1,301 @@
+  return auth.scheme === "Basic";
+}
+
+/**
+ * Type guard to check if authorization is OAuth type.
+ */
+export function isOAuth(auth: Authorization): auth is OAuth1Authorization {
+  return auth.scheme === "OAuth";
+}
+
+/**
+ * Parses key="value" parameters from a string.
+ * Used for OAuth and other schemes with comma-separated key-value pairs.
+ *
+ * @param parametersString The parameters string.
+ */
+function parseKeyValueParameters(
+  parametersString: string,
+): Record<string, string> {
+  const params: Record<string, string> = {};
+
+  // Match key="value" pairs
+  const paramPattern = /(\w+)="([^"]*)"/g;
+  let match = paramPattern.exec(parametersString);
+
+  while (match !== null) {
+    const [, key, value] = match;
+    if (key && value !== undefined) {
+      params[key] = value;
+    }
+    match = paramPattern.exec(parametersString);
+  }
+
+  return params;
+}
+
+/**
+ * Parses an Authorization header value into its scheme and parameters.
+ *
+ * @param authorization The Authorization header value.
+ * @throws {Error} If the authorization header is malformed.
+ *
+ * @example
+ * ```typescript
+ * const auth = parseAuthorization("Bearer token123");
+ * // { scheme: "Bearer", token: "token123" }
+ * ```
+ *
+ * @example
+ * ```typescript
+ * const auth = parseAuthorization("Basic dXNlcjpwYXNz");
+ * // { scheme: "Basic", credentials: "dXNlcjpwYXNz" }
+ * ```
+ *
+ * @example
+ * ```typescript
+ * const auth = parseAuthorization('OAuth oauth_consumer_key="key", oauth_token="token"');
+ * // { scheme: "OAuth", parameters: { oauth_consumer_key: "key", oauth_token: "token" } }
+ * ```
+ *
+ * @example
+ * ```typescript
+ * const auth = parseAuthorization('Digest realm="Example", nonce="abc"');
+ * // { scheme: "Digest", parametersRaw: 'realm="Example", nonce="abc"', parameters: { realm: "Example", nonce: "abc" } }
+ * ```
+ */
+export function parseAuthorization(authorization: string): Authorization {
+  const trimmed = authorization.trim();
+  const spaceIndex = trimmed.indexOf(" ");
+
+  if (spaceIndex === -1) {
+    throw new Error("Invalid Authorization header format");
+  }
+
+  const scheme = trimmed.slice(0, spaceIndex);
+  const parameters = trimmed.slice(spaceIndex + 1).trim();
+
+  if (scheme === "") {
+    throw new Error("Authorization scheme cannot be empty");
+  }
+
+  if (parameters === "") {
+    throw new Error("Authorization parameters cannot be empty");
+  }
+
+  // Return discriminated union based on scheme
+  if (scheme === "Bearer") {
+    return {
+      scheme: "Bearer",
+      token: parameters,
+    };
+  }
+
+  if (scheme === "Basic") {
+    return {
+      scheme: "Basic",
+      credentials: parameters,
+    };
+  }
+
+  if (scheme === "OAuth") {
+    return {
+      scheme: "OAuth",
+      parameters: parseKeyValueParameters(parameters) as OAuth1Parameters,
+    };
+  }
+
+  return {
+    scheme,
+    parametersRaw: parameters,
+    parameters: parseKeyValueParameters(parameters),
```

Comment:
I'd probably keep `parameters` as `string` and offer the possibility to parse on demand.
Something like
```
return { 
  scheme,
  parameters,
  parseKeyValues: () => parseKeyValueParameters(parameters)
}
```